ql/src/test/queries/clientpositive/udf_configurable.q

```diff
@@ -0,0 +1,55 @@
+CREATE TABLE test_udf_configurable (cint1 INT, cint2 INT, ctimestamp TIMESTAMP, text_timestamp STRING);
+INSERT INTO test_udf_configurable VALUES
+  (10000, 3, CAST('1970-01-01 01:02:03' AS TIMESTAMP), '1970-01-01 01:02:03 4'),
+  (20000, 5, CAST('1970-01-02 04:05:06' AS TIMESTAMP), '1970-01-02 04:05:06 5'),
+  (30000, 7, CAST('1970-01-03 07:08:09' AS TIMESTAMP), '1970-01-03 07:08:09 6');
+
+set hive.compat=latest;
+set hive.strict.timestamp.conversion=false;
+set hive.int.timestamp.conversion.in.seconds=true;
+set hive.local.time.zone=Asia/Bangkok;
+set hive.datetime.formatter=SIMPLE;
+set hive.masking.algo=sha512;
+set hive.use.googleregex.engine=true;
+
+-- On HiveServer2
+SELECT
+  -- DECIMAL because of hive.compat=latest
+  TYPEOF(cint1 / cint2),
+  -- Allowed by hive.strict.timestamp.conversion=false
+  cint1 = ctimestamp,
+  -- Allowed by hive.strict.timestamp.conversion=false
+  -- Interpreted as seconds because of hive.int.timestamp.conversion.in.seconds=true
+  CAST(cint1 AS TIMESTAMP),
+  -- The semantics of "u" is different between SimpleDateFormat and DateTimeFormatter
+  DATE_FORMAT(ctimestamp, 'yyyy-MM-dd HH:mm:ss u'),
+  FROM_UNIXTIME(cint1, 'yyyy-MM-dd HH:mm:ss u'),
+  TO_UNIX_TIMESTAMP(text_timestamp, 'yyyy-MM-dd HH:mm:ss u'),
+  UNIX_TIMESTAMP(text_timestamp, 'yyyy-MM-dd HH:mm:ss u'),
+  -- SHA512 is used
+  MASK_HASH(text_timestamp),
+  -- Java's Pattern doesn't support it, then it fails with hive.use.googleregex.engine=false
+  text_timestamp RLIKE '\\p{Katakana}+'
```

Comment:
The following files would touch SessionState or HiveConf.

```
git grep -l -e 'org.apache.hadoop.hive.ql.session.SessionState' -e 'org.apache.hadoop.hive.conf.HiveConf' | grep 'ql/src/java/org/apache/hadoop/hive/ql/udf/generic' 
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFAssertTrueOOM.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFBaseArithmetic.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFBaseCompare.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFBaseNumeric.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFCurrentAuthorizer.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFCurrentCatalog.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFCurrentDatabase.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFCurrentDate.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFCurrentGroups.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFCurrentSchema.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFCurrentTimestamp.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFCurrentUser.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFDateFormat.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFFromUnixTime.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFLoggedInUser.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFMaskHash.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFOPDivide.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFRegExp.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFRestrictInformationSchema.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFTimestamp.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFToUnixTimeStamp.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFUnixTimeStamp.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDTFGetSQLSchema.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDTFGetSplits.java
ql/src/java/org/apache/hadoop/hive/ql/udf/generic/InstantFormatter.java
```

I didn't add tests for the following UDFs because they are marked as "runtime constant" and the issue will unlikely happen. As another point, it is hard to configure some of them in qtest.

```
CURRENT_AUTHORIZER(),
CURRENT_CATALOG(),
CURRENT_DATABASE(),
CURRENT_DATE(),
CURRENT_GROUPS(),
CURRENT_SCHEMA(),
CURRENT_TIMESTAMP(),
CURRENT_USER(),
LOGGED_IN_USER(),
RESTRICT_INFORMATION_SCHEMA()
```